Content.Server/BodySystem/BodyManagerComponent.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Collections.Generic;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+
+namespace Robust.Shared.BodySystem {
+    public enum BodyPartCompatability { mechanical, biological, universal };	
+	public enum BodyPartType { other, torso, head, arm, hand, leg, foot };	  
+
+	public class BodyManagerComponent : Component {
+		private BodyTemplate _template;
+		private Dictionary<string, BodyPart> _parts;
+
+        public override string Name => "BodyManager";
+
+        public override void ExposeData(ObjectSerializer serializer) {
+            base.ExposeData(serializer);
+            //serializer.DataFieldCached(ref _heatResistance, "HeatResistance", 323);
+        }
+
+        public BodyManagerComponent(BodyTemplate _template){
+			this._template = _template;
+            _parts = new Dictionary<string, BodyPart>();
+		}
+		
+        /// <summary>
+        ///     Loads the given preset - forcefully sets all limbs on the attached entity to match it!
+        /// </summary>		
+		public void LoadBodyPreset(BodyPreset preset){
+			if(_template == null || preset.TemplateID != _template.ID) //Check that the preset's template and our template are equivalent.
+				return;
+			var prototypeManger = IoCManager.Resolve<IPrototypeManager>();
+			foreach(KeyValuePair<BodyPartType, string> slot in _template.Slots){ //Iterate through all limb slots in the template.
```

Comment:
You can do
```cs
foreach (var (type, slotName) in _template.Slots)
```
and then replace the usages of `slot.Key` and `Slot.Value`. Nice and compact :wink:.